src/session/session_api.c

```diff
@@ -619,6 +619,19 @@ __session_open_cursor(WT_SESSION *wt_session, const char *uri, WT_CURSOR *to_dup
     session = (WT_SESSION_IMPL *)wt_session;
     SESSION_API_CALL(session, open_cursor, config, cfg);
 
+    /*
+     * Check for early usage of a user session to collect statistics. If the connection is not fully
+     * ready but can be used, then only allow a cursor uri of "statistics:" only. The conditional is
+     * complicated. Allow the cursor to open if:
```

Comment:
"... to open if any of these conditions are met:"

(Otherwise, one might interpret all to be necessary).

Reply:
Fixed